### PR TITLE
Implement username picker

### PR DESCRIPTION
**Pull request by davidbojkovski**

### CODE REVIEW

File: app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameActivityTest.kt
Comment (makfazlic) [inline app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameActivityTest.kt:1]: Maybe just let me know how you use MockUsernames here?

File: app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameViewModelTest.kt
Comment (makfazlic) [inline app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameViewModelTest.kt:1]: I can see on line 34 you have username view model. Is this it?

File: app/src/main/java/com/github/mateo762/myapplication/ProcrastinotApplication.kt
Comment (makfazlic) [inline app/src/main/java/com/github/mateo762/myapplication/ProcrastinotApplication.kt:1]: Why is this deleted. Dont you need to initialize the notifications when the app starts? If now, where do you initialize the notification service from the previous sprint?

File: app/src/main/java/com/github/mateo762/myapplication/authentication/RegisterActivity.kt
Comment (makfazlic) [inline app/src/main/java/com/github/mateo762/myapplication/authentication/RegisterActivity.kt:1]: Why do you need the users map in the DB? 
Line: 55

**Review by makfazlic — APPROVED** [review]
Well done! A lot of work was completed in this Sprint by David. I left 5-6 questions I wish to get answered. See the comments. Overall, you have shown again that you can write clean and readable code, as I had no problem reviewing it. Also, great job on the Hilt injection.

File: app/src/androidTest/java/com/github/mateo762/myapplication/username/MockUsernameModule.kt
[the remark this answers is not in the document]
Comment (davidbojkovski) [reply app/src/androidTest/java/com/github/mateo762/myapplication/username/MockUsernameModule.kt:1]: I create a new database reference, list a new list, different from users. And all of the usernames are added there (for searching) and later when the user picks a username, it updates that list as well as the username of the user that we are in. This part only hardcodes the usernames references, which is different from users

File: app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameActivityTest.kt
Comment (davidbojkovski) [reply app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameActivityTest.kt:1]: It is automatically substituted in the previous part 
`
@Module
@TestInstallIn(
    components = [SingletonComponent::class],
    replaces = [UsernameModule::class]
)`

and you don't have to do anything in the activity

File: app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameViewModelTest.kt
Comment (davidbojkovski) [reply app/src/androidTest/java/com/github/mateo762/myapplication/username/UsernameViewModelTest.kt:1]: This is a separate test, and it is not connected with the previous activity test. Here an instance of ViewModel needs to be created and it takes a service as constructor parameters, so I have to inject the service, but it will inject the mock service that I created with the hardcoded values

File: app/src/main/java/com/github/mateo762/myapplication/ProcrastinotApplication.kt
Comment (davidbojkovski) [reply app/src/main/java/com/github/mateo762/myapplication/ProcrastinotApplication.kt:1]: There was a problem with the Hilt inject in the tests on Cirrus, so I moved this logic to the NotificationService, the one that sends the notifications

File: app/src/main/java/com/github/mateo762/myapplication/authentication/RegisterActivity.kt
Comment (davidbojkovski) [reply app/src/main/java/com/github/mateo762/myapplication/authentication/RegisterActivity.kt:1]: That is actually from before and not my implementation
